scalableminds/webknossos
# Add missing info in datasource-properties.json to Zarr v2 route

The wklibs use the zarrv2 `datasource-properties.json` route when opening the remote dataset. When this route is called the backend transforms layers to zarr layer. During the conversion process, some properties of layers were omitted. I updated the code to include the parameters that were not passed previously. 
For reference: The conversion code is called here in line 104:
https://github.com/scalableminds/webknossos/blob/97ef1d16c8ae2bdfad62f28808f5f237145aacb3/webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/ZarrStreamingController.scala#L94-L109

### URL of deployed dev instance (used for testing):
- https://addndinfotozarrstreamingdatasource.webknossos.xyz/

### Steps to test:
- Get a token to view the following 4d dataset: https://addndinfotozarrstreamingdatasource.webknossos.xyz/datasets/sample_organization/13457227-4d-mag4.local.zarr/view#256,256,17,0,0.826,pos-t=8
  - eg via: view the dataset
  - in the navbar menu select share and copy the link
  - open the link in a new private browser tab (where you are not logged in)
  - Copy the token from the URL params `token=<token>`  
- Check the following datasource-properties.json [https://addndinfotozarrstreamingdatasource.webknossos.xyz/data/zarr/sample_organization/13457227-4d-mag4.local.zarr/datasource-properties.json?token=<my-token>](https://addndinfotozarrstreamingdatasource.webknossos.xyz/data/zarr/sample_organization/13457227-4d-mag4.local.zarr/datasource-properties.json?token=<my-token>). It should include additional axes.

### TODOs:
- [x] Setup dev instance with an nd dataset with default view configs. Both should show in the datasource-properties.json returned by the route :)
- [x] Fix loading nd data chunks


### Issues:
- Needed for but does not solve [#7279](https://github.com/scalableminds/webknossos/issues/7279)

------
(Please delete unneeded items, merge only when none are left open)
